internal/pkg/admission/policy-server-pod-disruption-budget.go

```diff
@@ -0,0 +1,72 @@
+package admission
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/kubewarden/kubewarden-controller/internal/pkg/constants"
+	policiesv1 "github.com/kubewarden/kubewarden-controller/pkg/apis/policies/v1"
+	k8spoliciesv1 "k8s.io/api/policy/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *Reconciler) reconcilePolicyServerPodDisruptionBudget(ctx context.Context, policyServer *policiesv1.PolicyServer) error {
+	if policyServer.Spec.MinAvailable != nil || policyServer.Spec.MaxUnavailable != nil {
+		return reconcilePodDisruptionBudget(ctx, policyServer, r.Client, r.DeploymentsNamespace)
+	}
+	return deletePodDisruptionBudget(ctx, policyServer, r.Client, r.DeploymentsNamespace)
+}
+
+func deletePodDisruptionBudget(ctx context.Context, policyServer *policiesv1.PolicyServer, k8s client.Client, namespace string) error {
+	pdb := &k8spoliciesv1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      buildPodDisruptionBudgetName(policyServer),
+			Namespace: namespace,
+		},
+	}
+	err := client.IgnoreNotFound(k8s.Delete(ctx, pdb))
+	if err != nil {
+		err = errors.Join(fmt.Errorf("failed to delete PodDisruptionBudget"), err)
+	}
+	return err
+}
+
+func buildPodDisruptionBudgetName(policyServer *policiesv1.PolicyServer) string {
+	return fmt.Sprintf("%s-%s", policyServer.NameWithPrefix(), constants.PolicyServerPodDisruptionBudgetSuffix)
+}
+
+func reconcilePodDisruptionBudget(ctx context.Context, policyServer *policiesv1.PolicyServer, k8s client.Client, namespace string) error {
+	pdb := &k8spoliciesv1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      buildPodDisruptionBudgetName(policyServer),
+			Namespace: namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, k8s, pdb, func() error {
+		pdb.Name = buildPodDisruptionBudgetName(policyServer)
+		pdb.Namespace = namespace
+		pdb.OwnerReferences = []metav1.OwnerReference{
+			*metav1.NewControllerRef(policyServer, policiesv1.GroupVersion.WithKind("PolicyServer")),
+		}
+		pdb.Spec.Selector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				constants.AppLabelKey:          policyServer.AppLabel(),
+				constants.PolicyServerLabelKey: policyServer.GetName(),
+			},
+		}
+		if policyServer.Spec.MinAvailable != nil {
+			pdb.Spec.MinAvailable = policyServer.Spec.MinAvailable
+		} else {
+			pdb.Spec.MaxUnavailable = policyServer.Spec.MaxUnavailable
+		}
+		return nil
+	})
```

Comment:
The part where we set the ownerRef can be handled in a nicer way using [`controllerutil.SetOwnerReference`](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.17.2/pkg/controller/controllerutil#SetOwnerReference).

For example:
```golang
secret := &corev1.Secret{
			TypeMeta: metav1.TypeMeta{
				Kind:       "Secret",
				APIVersion: "v1",
			},
			ObjectMeta: metav1.ObjectMeta{
				Namespace: app.ObjectMeta.Namespace,
				Name:      generatedRuntimeConfigSecretName,
				Labels: map[string]string{
					spinapp.NameLabelKey: app.ObjectMeta.Name,
				},
			},
			Data: map[string][]byte{
				"runtime-config.toml": tomlValue,
			},
		}
		err = controllerutil.SetOwnerReference(app, secret, r.Scheme)
		if err != nil {
			return fmt.Errorf("failed to set runtimeconfig owner reference: %w", err)
		}

		err = r.Client.Create(ctx, secret)
```

this is taken from [here](https://github.com/spinkube/spin-operator/blob/9ffa340b1e057a47f9fb8c2eaf8edf675f377db0/internal/controller/spinapp_controller.go#L252C3-L273C37)